search-commons/src/test/java/no/unit/nva/search/models/SearchResponseDtoTest.java

```diff
@@ -38,6 +49,24 @@ void jsonSerializationShouldKeepDeprecatedFieldsUntilFrontendHasMigrated() throw
         assertThat(json, is(jsonObject().where("total", JsonMatchers.jsonLong(searchResponse.getSize()))));
     }
 
+    @Test
+    void searchResponseShouldMapSnakeCaseToCamelCase() throws IOException {
+        var searchResponse = getSearchResponse();
+        var aggregations = SearchResponseDto.fromSearchResponse(searchResponse, randomUri()).getAggregations();
+
+        var docCount = aggregations.findValue("docCount");
+        Assertions.assertNotNull(docCount);
+        Assertions.assertEquals(2, docCount.asInt());
+
+        var snakeCase = aggregations.findValue("doc_cound_error_upper_bound");
+        assertThat(snakeCase, is(nullValue()));
+    }
+
+    private SearchResponse getSearchResponse() throws IOException {
+        String jsonResponse = stringFromResources(Path.of(SAMPLE_OPENSEARCH_RESPONSE_JSON));
+        return SearchResponseUtil.getSearchResponseFromJson(jsonResponse);
+    }
+
 
     private SearchResponseDto randomResponse() {
         return SearchResponseDto.builder()
```

Comment:
There should also be a test at the lambda layer that makes sure that the return value has the fields we have "promised" to return to the frontend